0x6b Information Gathering on iOS

cpholguera opened the pull request:
This PR covers issue #1300 and #1249.

sushi2k commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -1005,12 +1236,12 @@ You should now be able to reach Burp on your iOS device. Open Safari on iOS and
 1. Configure your Proxy by selecting Manual
 1. Type in 127.0.0.1 as Server
-1. Type in 8080 as Port
+2. Type in 8080 as Port
```
this should be 1.

cpholguera replied:
done

sushi2k commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -1005,12 +1236,12 @@ You should now be able to reach Burp on your iOS device. Open Safari on iOS and
 
 1. Go to Settings
-1. Wi-Fi
+2. Wi-Fi
```
this should be 1.

cpholguera replied:
done

commjoen commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -786,42 +757,298 @@ Possible values for the property [UIDeviceFamily](https://developer.apple.com/li
+
+##### Monitoring System Logs
+
+MYTODO create.
```
dont' forget about it 👍

cpholguera replied:
done, I've moved existing content from another section and included passionfruit

commjoen commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -786,42 +757,298 @@ Possible values for the property [UIDeviceFamily](https://developer.apple.com/li
+
+MobSF is a penetration testing framework that is capable of analyzing IPA files and can be used before even installing the app on your testing device.
+
+Once you have MobSF up and running you can open it in your browser by navigating to <http://127.0.0.1:8000>. Simply select the IPA you want to analyse and MobSF will start its job. The bigger the app the longer it takes, but usually you should get some feedback within a few minutes.
```
note: this does depend on how you run it: if you run it locally , then yes, if you run it in the docker container, it depends on the configuration. Do we explain that somewhere?

cpholguera replied:
actually I moved this now to Recommended Tools - MobSF. There it is shown how to launch it with docker and the command includes exposing port 8000

cpholguera commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -786,13 +757,288 @@ Possible values for the property [UIDeviceFamily](https://developer.apple.com/li
+The most relevant items are:
+
+- `Info.plist` contains configuration information for the application, such as its bundle ID, version number, and display name.
+- resources such as images/icons, localized content (`<language>.lproj`), text files, audio files, etc.
```
now is Various resources... includes 2 items of the list together

cpholguera commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -786,13 +757,288 @@ Possible values for the property [UIDeviceFamily](https://developer.apple.com/li
+
+###### The Info.plist File
+
+The information property list or `Info.plist` (named by convention) is main source of information for an iOS app. It consists of a structured file containing key-value pairs describing essential configuration information about the app. Actually, all bundled executables (plug-ins, frameworks, and apps) are expected to have an `Info.plist` file. You can find all possible key in the [Apple Developer ...
```
done

cpholguera commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -786,42 +757,298 @@ Possible values for the property [UIDeviceFamily](https://developer.apple.com/li
+
+###### Other App Resources
+
+It is normally worth taking a look at the rest of the resources and files that you may find in the Application Bundle (.app) inside the IPA as some times they contain additional goodies like encrypted databases, certificates, etc.
```
I leave it like it is because:

> if "etc." is used in the middle of a sentence, it is followed by a comma. (Tennis, soccer, baseball, etc., are outdoor games.) However, if this word appears at the end of a sentence then the period (which is part of "etc.") serves as the final punctuation mark.

cpholguera commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -786,42 +757,298 @@ Possible values for the property [UIDeviceFamily](https://developer.apple.com/li
+
+###### Native Libraries
+
+iOS native libraries are known as Frameworks. You can inspect them by listing the `Frameworks` folder in the IPA:
```
I think you refer to 0x6i? I've added this sentence:

Please note that this might not be the complete list of third party libraries being used by the app as some can be part of the source code, meaning that they'll be compiled in the app binary and therefore cannot be found as standalone libraries or Frameworks in the `Frameworks` folder.

OMTGreleaser commented:
Results for commit 42162cedb26ffe39f21772c34ab7e979161d473f: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

OMTGreleaser commented:
Results for commit d07e6e3b77f7fbfda53a0d67cdef543497c173d5: Broken link result: [!] Error(s) found by the Links validator: 7 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

OMTGreleaser commented:
Results for commit cb3d26a87bdb93ff0526d4bd4b2f35d3d9b15192: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

commjoen commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -786,61 +784,322 @@ Possible values for the property [UIDeviceFamily](https://developer.apple.com/li
+
+- Which files are included in the package?
+- Which Frameworks does the app use?
+- Which permissions does the app request to the user and for what reason?
```
Don't forget capabilities registered.

cpholguera replied:
done

commjoen commented on Document/0x06b-Basic-Security-Testing.md:
```diff
@@ -215,6 +215,14 @@ $ docker pull opensecurity/mobile-security-framework-mobsf
+
+After MobSF is done with its analysis, you will receive a one-page overview of all the tests that were executed. While it may look daunting at first, the page is split up into multiple sections, each with their own purpose. Together, all the sections give a good first indication of the attack surface of the application. You can also execute additional actions, such as:
+
+- Download a class-dump, if the app was written in Objective-C; if it is written in Swift no classdump can be created.
```
Have you tried running outside the docker container? You get better results for both swift and objc if you run it just with python. Try it :)

cpholguera replied:
I've added both options now and put a comment for that: "> By running it locally on a macOS host you'll benefit from a slightly better class-dump output."

The output is a bit cleaner in fact.

OMTGreleaser commented:
Results for commit ab37c5ef32e01bc032c8e367b37283fc6d86b60d: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

OMTGreleaser commented:
Results for commit 104c9de86bd76c3032d9d6681c0dd24f50e13edc: Broken link result: [!] Error(s) found by the Links validator: 2 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

cpholguera commented:
> Some comments still to do. What happened to soCAT to monitor syslog for system interaction? That should be part of this...

The socat method was tested and is now also part of the monitoring logs section. 

also moved the already existing "logs via needle" description and removed duplicated instructions on iOS logs from other chapters.

OMTGreleaser commented:
Results for commit bd2716881d5b747c95cd9ab90300d425d7ade116: Broken link result: [!] Error(s) found by the Links validator: 3 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        1 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.
